profileservice/api/controllers/profiles.py

```diff
@@ -304,72 +313,22 @@ def pii_post():
-
-        # if it is an existing entry, then check if the information in db matches with the id token
-        auth_pass = check_auth(dataset, tk_uin, tk_phone, tk_is_uin, tk_is_phone)
-    except:
-        dataset = None
-        is_new_entry = True
-
-    # check if the email already exists
-    if tk_is_uin:
-        try:
-            dataset = mongoutils.get_pii_dataset_from_field('uin', tk_uin)
-            # if there is a dataset, it means that the email is existing in the database
-            if dataset is not None:
-                # ToDo Following lines will be commented out due to the following assumption
-                # that the email in the database doesn't get updated so, no change in email.
-                # However, if there is email update available, the following part should be revived.
-                # # check if the id token and db info matches
-                # if not (auth_pass):
-                #     msg = {
-                #         "reason": "The user info in id token and db are not matching.",
-                #         "error": "Authorization Failed."
-                #     }
-                #     msg_json = jsonutils.create_log_json("PII", "POST", msg)
-                #     self.logger.error("PII POST " + json.dumps(msg_json))
-                #     return jsonutils.create_auth_fail_message()
-                #
-                # if not (auth_pass):
-                #     msg = {
-                #         "reason": "The user info in id token and db are not matching.",
-                #         "error": "Authorization Failed."
-                #     }
-                #     msg_json = jsonutils.create_log_json("PII", "POST", msg)
-                #     self.logger.error("PII POST " + json.dumps(msg_json))
-                #     return jsonutils.create_auth_fail_message()
-
-                pid = dataset.get_pid()
-                non_pii_uuid_from_dataset = dataset.uuid
-                dataset = append_non_pii_uuid(non_pii_uuid, non_pii_uuid_from_dataset, dataset)
-                currenttime = otherutils.get_current_time_utc()
-                dataset.set_last_modified_date(currenttime)
-                result, pii_dataset = mongoutils.update_pii_dataset_in_mongo_by_field(cfg.FIELD_PID, pid, dataset)
-                msg = {
-                    "reason": "UIN already exists: " + str(pid),
-                    "warning": "UIN already exists: " + request.url,
-                }
-                msg_json = jsonutils.create_log_json("PII", "POST", msg)
-                logging.warning("PII POST " + json.dumps(msg_json))
-                return rs_handlers.return_id('UIN already exists.', 'pid', pid)
-        except:
-            return rs_handlers.internal_server_error()
-
-    # check if the phonenumber already exists
-    if tk_is_phone:
-        try:
-            dataset = mongoutils.get_pii_dataset_from_field('phone', tk_phone)
-
+        dataset = mongoutils.get_pii_dataset_from_field('uid', tk_uid)
```

Comment:
Since currently `uid` field is not in the database, this will not return profiles that are already in the database. Also, this might eventually result in duplicate PII records. One option would be to set this key name as a config option with the default value being 'uin'. Please also see if there are other better solutions. I would like to consider updating the database as a last resort.

Reply:
@shurwit, tagging in case you missed this comment above. Thanks.